### PR TITLE
Check for HTTP error when downloading media, other error tweaks

### LINKED ISSUE #100: Raise exception instead of log.error when possible
We should go through the uses of `log.error` and convert them to raise exceptions, remove them entirely, or leave as is depending on context. For example https://github.com/mitodl/ocw-data-parser/blob/32a9b125c6a5bdcb1af38dafd7a82ca7407cadc9/ocw_data_parser/ocw_data_parser.py#L57 should be removed entirely since loading a JSON should always work or throw an exception.

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #100 

#### What's this PR do?
We should call `.raise_for_status` for all requests made so a 404 response will cause an exception instead of an empty file.

#### How should this be manually tested?
Nothing should change in the output

### CODE REVIEW

File: ocw_data_parser/ocw_data_parser.py [hunk 1]
```diff
@@ -587,8 +588,9 @@ def extract_foreign_media_locally(self):
         os.makedirs(path_to_containing_folder, exist_ok=True)
         for media in self.large_media_links:
             file_name = media["link"].split("/")[-1]
+            response = requests.get(media["link"])
+            response.raise_for_status()
```
Comment (mbertrand) [inline ocw_data_parser/ocw_data_parser.py]: Should the parsing be halted by an error if one of these links is no longer valid?  Maybe it should be in a try-catch block with exception logging [like below](https://github.com/mitodl/ocw-data-parser/pull/106/files#diff-893fe7ed068c91774e4c9132fefab09e62254d231c461f7debb7b6d70ce5ff59R793-R798)?

**Review by mbertrand — COMMENTED** [review]
Looks great overall, just 1 question.

File: ocw_data_parser/ocw_data_parser.py (hunk 1 above)
Comment (noisecapella) [reply ocw_data_parser/ocw_data_parser.py]: Sounds good, I just made the change

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**